.github/workflows/on-push-main.yaml

```diff
@@ -0,0 +1,39 @@
+---
+name: "On-Push-Main"
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+permissions:
+  contents: read
+  packages: write
+  attestations: write
+  id-token: write
+  pull-requests: write
+  pages: write
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    outputs:
+      release_created: ${{ steps.release.outputs.release_created }}
+      tag_name: ${{ steps.release.outputs.tag_name }}
+    steps:
+      - id: release
+        uses: googleapis/release-please-action@v4
+        with:
+          token: ${{ secrets.RELEASE_PLEASE_GH_TOKEN }}
+          config-file: .github/release-please-config.json
+          manifest-file: .github/.release-please-manifest.json
+  build-and-push:
+    needs: release-please
+    uses: ./.github/workflows/reusable__upload-to-ghcr.yaml
+    with:
+      context: "."
+      image_name: "service"
+      version: ${{needs.release-please.outputs.tag_name}}
+  documentation:
+    uses: ./.github/workflows/reusable__documentation.yaml
+    with:
+      COMMIT_HASH: ${{ github.sha }}
+      DEPLOY: ${{ github.ref_name == 'main' }}
```

Comment:
you don't need that check anymore
just use true
or remove that check in the child workflow at all